join.sh

```diff
@@ -16,10 +23,16 @@ cat <<EOS > /etc/docker/daemon.json
 }
 EOS
 
+#Configuration for Docker Behind a Proxy
+if [ $USE_PROXY == true ]; then
+  mkdir -p /etc/systemd/system/docker.service.d
+  CreateProxyConfiguraton
+fi
+
 export DEBIAN_FRONTEND=noninteractive
 apt-get update -qq
 apt-get install -y -qq --no-install-recommends \
-    docker.io \
+    docker.io=$DOCKER_VERSION \
```

Comment:
We will definitely want to support that, since some users will certainly want to run their own image repositories.

I wonder if we can simply use an environment variable for the full `<package>=<version>` string and then only worry about something more complex once we find we need to install multiple packages?